jiant/models.py

```diff
@@ -1196,3 +1223,26 @@ def get_elmo_mixing_weights(self, tasks=[]):
                         self.sent_encoder._text_field_embedder, task=None
                     )
         return params
+
+
+def input_module_uses_pair_embedding(input_module):
+    """
+    This function tells whether the input module concatenate the two sentences in a pair when
+    running on pair tasks, like what GPT / BERT do on MNLI.
+    It seems redundant now, but it allows us to load similar models from other sources later on
+    """
+    from jiant.pytorch_transformers_interface import input_module_uses_pytorch_transformers
+
+    return input_module_uses_pytorch_transformers(input_module)
+
+
+def input_module_uses_mirrored_pair(input_module):
+    """
+    This function tells whether the input model uses raw pair and mirrored pair simutaneously when
```

Comment:
Explain what 'mirrored pair' is. Is this something like A [sep] B [sep] A?

Reply:
It's a trick gpt do when predicting how similar two sentences are. If the sentences are A and B. It will create two inputs "[start] A [sep] B [end]" and "[start] B [sep] A [end]", and run the model on both of them, then sum up the logits, this allow one directional models like gpt also evaluate the similarity of the pair from both sides.